tests/test-github-policy.sh

```
@@ -550,6 +554,51 @@ grep -q "not-yet-there.yml does not exist at pinned SHA" "$TMP_DIR/two-workflows
+[ ! -f "$TMP_DIR/state/auto-merge" ] || fail "auto-merge was not restored after the organization deletion failed"
+ok "partial cleanup still attempts every independent step and names the failure"
+rm -f "$TMP_DIR/state/ruleset.json"
+# A companion ruleset with no organization ruleset is an interrupted adoption, not a bare repository.
+rm -f "$TMP_DIR/state/branch.json" "$TMP_DIR/state/ruleset.json" "$TMP_DIR/state/auto-merge" "$TMP_DIR/state/bad-effective-used"
+jq '.managedRepositoryRuleset + {id:321}' "$POLICY" >"$TMP_DIR/state/repo-ruleset.json"
+if run_policy apply "$POLICY" >"$TMP_DIR/bootstrap-companion.out" 2>&1; then
+  fail "a companion-only repository was bootstrapped over"
+fi
+grep -q "companion repository ruleset already exists" "$TMP_DIR/bootstrap-companion.out" \
+  || fail "companion-only state was not routed to manual recovery: $(tail -1 "$TMP_DIR/bootstrap-companion.out")"
+[ -f "$TMP_DIR/state/repo-ruleset.json" ] && ok "companion-only state refused and left untouched" || fail "companion ruleset was deleted"
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)